executor/common_fuchsia.h

```diff
@@ -43,7 +43,7 @@ static void segv_handler(void)
 		longjmp(segv_env, 1);
 	}
 	debug("recover: exiting\n");
-	doexit(SIGSEGV);
+	doexit_thread(SIGSEGV);
```

Comment:
Why is this change necessary? What's wrong with doexit? We need to terminate the whole process here, so doexit looks like the right thing.

Reply:
Oh yes, I was getting an error with doexit and doexit_thread fixed it.
I will try again to see if I can fix this.

Reply:
When building fuchsia image with `doexit` the build fail with `error: unused function 'doexit_thread' [-Werror, -Wunused-function]`. 
To fix this I can skip the declaration of `doexit_thread` inside for fuchsia: 
inside executor.cc
```
#if !GOOS_fuchsia
static NORETURN void doexit_thread(int status);
#endif
```
and inside common.h
```
#if !GOOS_fuchsia
NORETURN void doexit_thread(int status)
{
	// For BSD systems, _exit seems to do exactly what's needed.
	doexit(status);
}
#endif
 ```

But, since `doexit_thread` is just a wrapper around doexit, I think we can keep it like this. What do you think @dvyukov ?
https://github.com/behouba/syzkaller/blob/3aa720cd1c82d87f6894be58f36d5c03b3da5c16/executor/common.h#L54-L64

Reply:
I would prefer `#if !GOOS_fuchsia`. It looks more reliable with respect to future changes. The fact that doexit and doexit_thread do the same is just an implementation detail, in future they may do different things for fuchsia.

Reply:
Ok, thank you! 
I will add a new commit.